cmake/FindCaDiCaL.cmake

```diff
@@ -13,15 +13,80 @@
 # CaDiCaL_INCLUDE_DIR - the CaDiCaL include directory
 # CaDiCaL_LIBRARIES - Libraries needed to use CaDiCaL
 
+include(deps-helper)
+
 find_path(CaDiCaL_INCLUDE_DIR NAMES cadical.hpp)
 find_library(CaDiCaL_LIBRARIES NAMES cadical)
 
-include(FindPackageHandleStandardArgs)
-find_package_handle_standard_args(CaDiCaL
-  DEFAULT_MSG
-  CaDiCaL_INCLUDE_DIR CaDiCaL_LIBRARIES)
+set(CaDiCaL_FOUND_SYSTEM FALSE)
+if(CaDiCaL_INCLUDE_DIR AND CaDiCaL_LIBRARIES)
+    set(CaDiCaL_FOUND_SYSTEM TRUE)
+
+    # Unfortunately it is not part of the headers
+    find_library(CaDiCaL_BINARY NAMES cadical)
+    if(CaDiCaL_BINARY)
+      execute_process(
+        COMMAND ${CaDiCaL_BINARY} --version
+        OUTPUT_VARIALE CaDiCaL_VERSION
+      )
+    else()
+      set(CaDiCaL_VERSION "")
+    endif()
+
+    check_system_version("CaDiCaL")
+endif()
+
+if(NOT CaDiCaL_FOUND_SYSTEM)
+    include(CheckSymbolExists)
+    include(ExternalProject)
+
+    fail_if_include_missing("sys/resource.h" "CaDiCaL")
+
+    # TODO(mpreiner): Use a more recent version?
```

Comment:
Yes, but we'll do that in a separate commit (we'll do that when preparing for SMT-COMP).